eva-accession-import-automation/generate_properties.py

```diff
@@ -0,0 +1,168 @@
+"""spring.batch.job.names=IMPORT_DBSNP_VARIANTS_JOB
+
+accessioning.instanceId=instance-import
+accessioning.variant.categoryId=ss
+accessioning.monotonic.ss.nextBlockInterval=10000
+accessioning.monotonic.ss.blockStartValue=10000
+accessioning.monotonic.ss.blockSize=1000
+
+parameters.assemblyAccession={args.assembly_accession}
+parameters.assemblyName={args.assembly_name}
+parameters.assemblyReportUrl=file:{args.assembly_report}
+parameters.taxonomyAccession={args.taxonomy}
+parameters.chunkSize=1000
+parameters.pageSize=100
+#parameters.forceRestart=true
+#parameters.forceImport=false
+parameters.fasta={args.fasta}
+{args.optional_build_line}
+
+dbsnp.datasource.driver-class-name=org.postgresql.Driver
+dbsnp.datasource.url=jdbc:postgresql://{args.dbsnp_host}:{args.dbsnp_port}/dbsnp_{args.dbsnp_build}?currentSchema=dbsnp_{args.database_name}
+dbsnp.datasource.username={args.dbsnp_user}
+dbsnp.datasource.password={args.dbsnp_password}
+dbsnp.datasource.tomcat.max-active=3
+
+spring.datasource.driver-class-name=org.postgresql.Driver
+spring.datasource.url=jdbc:postgresql://{args.job_tracker_host}:{args.job_tracker_port}/{args.job_tracker_db}
+spring.datasource.username={args.job_tracker_user}
+spring.datasource.password={args.job_tracker_password}
+spring.datasource.tomcat.max-active=3
+spring.jpa.generate-ddl=true
+
+spring.data.mongodb.host={args.mongo_host}
+spring.data.mongodb.port={args.mongo_port}
+spring.data.mongodb.database={args.mongo_acc_db}
+spring.data.mongodb.username={args.mongo_user}
+spring.data.mongodb.password={args.mongo_password}
+spring.data.mongodb.authentication-database={args.mongo_auth_db}
+mongodb.read-preference=primaryPreferred
+
+spring.main.web-environment=false
+
+logging.level.uk.ac.ebi.eva.accession.dbsnp=DEBUG
+logging.level.uk.ac.ebi.eva.accession.dbsnp.listeners=INFO
+logging.level.org.springframework.jdbc.datasource=DEBUG
+"""
+    properties_filename = "{}_{}.properties".format(args.build, args.assembly_accession)
+    with open(properties_filename, "w") as properties_file:
+        properties_file.write(template.format(args=args))
+
+
+def get_user_and_password_and_port_from_pgpass_for_host(pg_host):
+    with open(expanduser("~/.pgpass")) as pgpass_file:
+        for line in pgpass_file:
+            if line.split(':', 1)[0] == pg_host:
+                fields = line.rstrip('\n').split(':')
+                return fields[3], fields[4], fields[1]
+
+
+def init_logger():
+    logging.basicConfig(level=logging.INFO, format='%(asctime)-15s %(levelname)s %(message)s')
+    result_logger = logging.getLogger(__name__)
+    return result_logger
+
+
+if __name__ == "__main__":
+    logger = init_logger()
+
+    parser = argparse.ArgumentParser(
+        description='Generate a properties file that can be used for eva-accession-import',
+        add_help=False)
+    parser.add_argument("-b", "--build", help="dbSNP build number, e.g. 151", required=True)
+    parser.add_argument("-l", "--latest-build",
```

Comment:
I would prefer to have the flag `previous-build` so if the flag is not specified we will assume to be generating the properties for the latest build (more common case)

Reply:
agreed offline to keep it as it is